packages/headless/src/controllers/core/did-you-mean/headless-core-did-you-mean.ts

```diff
@@ -93,6 +98,7 @@ export function buildCoreDidYouMean(engine: CoreEngine): DidYouMean {
         hasQueryCorrection:
           state.didYouMean.queryCorrection.correctedQuery !== '' ||
           state.didYouMean.wasCorrectedTo !== '',
+        automaticallyCorrectQuery: state.didYouMean.automaticallyCorrectQuery,
```

Comment:
I don't think we should change the state of the controller we return.
Also, there should probably be a new option on the controller.

1- The state we return from a controller is meant to be used by UI piece to control the rendering logic. This `automaticallyCorrectQuery` does not seem relevant: A good hint is the fact that there was no UI change in Atomic, for example.

2- To control the `enableAutomaticQueryCorrection()`, it would be much better from an implementer POV if they could simply pass an option on the controller. And when the controller is built, we check what's the option value, and dispatch the appropriate `disableAutomaticQueryCorrection` action.